weave/wandb_interface/wandb_lite_run.py

```diff
@@ -1,19 +1,45 @@
-import os
-import logging
 import datetime
-import json
+import logging
+import platform
 import typing
+
+from wandb import __version__, util
 from wandb import errors as wandb_errors
-from wandb.apis.public import Run
-from wandb.sdk.internal.file_pusher import FilePusher
-from wandb.sdk.internal import file_stream
-from wandb.sdk.internal.internal_api import Api as InternalApi
-from wandb.sdk.lib import runid
-from weave import wandb_client_api
-from weave import errors
+from wandb.apis import public
+
+from .artifacts import artifact_saver
+from .internal import file_stream
+from .internal.file_pusher import FilePusher
+from .internal.internal_api import Api as InternalApi
+from .internal.thread_local_settings import _thread_local_api_settings
+from .lib import config_util, runid
+from .lib.ipython import _get_python_type
+
+if typing.TYPE_CHECKING:
+    from wandb.sdk.artifacts.artifact import Artifact
 
 logger = logging.getLogger(__name__)
 
+_p_api = None
+
+
+def wandb_public_api() -> public.Api:
+    global _p_api
+    if _p_api is None:
+        _p_api = public.Api(timeout=30)
+    return _p_api
```

Comment:
Made this a global to re-use the http session.

Reply:
i am not sure how this will impact authentication. If _p_api is global, then i think we will always use the first instance, which will be authed with the creds of the first caller (when running in server mode)

Reply:
i need to double check if any auth state is persisted on the object itself

Reply:
also, why extract from the weave.wandb_client_api ?

Reply:
please ignore the above.

Reply:
This is fine when used in wandb, but if used in weave, we cannot have a global due to auth bleed